### DISCUSSION TITLE
Episode 149: CityStrides.com, Graph Algorithms and More!

### DISCUSSION THREAD

**Discussion by codereport** [codereport/adsp2#40] [Episode]
Episode Link: https://adspthepodcast.com/2023/09/29/Episode-149.html

Feel free to write comments or ask questions 🙂

**Comment by jefftrull** [comment]
I think CityStrides is the Clustered Orienteering Problem FWIW

**Reply by jefftrull** [reply]
Seriously, take a look. A "cluster" of vertices has to be all visited to count towards success. The edges are used to determine time spent and the problem is to optimize the number of clusters (streets) visited given an upper limit in time. It fits the problem.

**Comment by ladnir** [comment]
You also might want to do a preprocessing where you [contract](https://en.m.wikipedia.org/wiki/Edge_contraction)  one of the edges that are connected to a degree 2 vertex. Ie for a vertex that has only one "in" edge and one "out" edge, then the first or second edge is contacted. As you mentioned, these edges complicate the algorithm but don't have any additional meaning. 

Additionally, you told Bryce that the weight idea doesn't make sense because of these extra edges. I would argue once you contract these extra edges, you could add weights to the edges that encode the distance. And then you could the do a greed optimizing based on these weight.

Cheers
